include/crocoddyl/multibody/residuals/contact-force.hxx

```diff
@@ -40,6 +40,9 @@ void ResidualModelContactForceTpl<Scalar>::calc(const boost::shared_ptr<Residual
 
   // We transform the force to the contact frame
   switch (d->contact_type) {
+    case Contact1D:
+      data->r = Eigen::Matrix<Scalar, 1, 1>((d->contact->jMf.actInv(d->contact->f) - fref_).linear()[2]);
```

Comment:
Why are you using `Eigen::Matrix<Scalar, 1, 1>`?

I think it will allocate memory.

Reply:
This is on the stack.